### PR TITLE
builtin/aws/ecs: properly clamp the service name. Closes #873

### PR DESCRIPTION
**Pull request by evanphx**
The old code would panic (see #873), this new version achieves the aim of the existing code, but in a bonus move, doesn't panic at all!

### CODE REVIEW

File: builtin/aws/ecs/platform.go [hunk 1]
```diff
@@ -773,9 +773,14 @@ func (p *Platform) Launch(
...
 	s.Update("Registered Task definition: %s", family)
-	rand := id[len(id)-(31-len(app.App)):]
 
-	serviceName := fmt.Sprintf("%s-%s", app.App, rand)
+	serviceName := fmt.Sprintf("%s-%s", app.App, id)
+
+	// We have to clamp at a length of 32 because the Name field to CreateTargetGroup
+	// requires that the name is 32 characters or less.
+	if len(serviceName) > 32 {
+		serviceName = serviceName[:32]
```
Comment (briancain) [inline builtin/aws/ecs/platform.go:782]: Maybe worth logging that the name is getting trimmed here due to a limitation in `CreateTargetGroup`?

```suggestion
                L.Info("Trimming service name to 32 characters due to limitation in 'CreateTargetGroup' name field")
		serviceName = serviceName[:32]
```

**Review by briancain — APPROVED** [review]
Hurray for no panics! :tada:

File: builtin/aws/ecs/platform.go (hunk 1 above)
Comment (evanphx) [reply builtin/aws/ecs/platform.go:782]: Nah, because the user doesn't really need to even know about this value as it's just fed to the targetgroup name.

Comment (briancain) [reply builtin/aws/ecs/platform.go:782]: Ahhh, cool :+1: Yeah, if they don't know about this value they probably won't care that it got trimmed

**Review by briancain — APPROVED** [review]
Hurray for less panics!

**Review by krantzinator — APPROVED** [review]
_whack_

**Comment by evanphx** [comment]
> Hurray for less panics!
_____________^^^^^_______ - Syntax error detected, perhaps you meant _fewer_?

**Pull request merged**